Dockerfile

```diff
@@ -1,25 +1,35 @@
-FROM python:3.6-alpine
-
-RUN apk update && apk upgrade
-RUN apk add --no-cache nmap nmap-scripts
-RUN apk add --no-cache libpng freetype libstdc++ pkgconfig
-RUN apk add --no-cache --virtual .build-deps gcc build-base python-dev libpng-dev musl-dev freetype-dev
-RUN ln -s /usr/include/locale.h /usr/include/xlocale.h
-
-COPY requirements.txt /code/requirements.txt
-RUN pip install -r /code/requirements.txt
-
-RUN apk del .build-deps
-
-ENV PYTHONPATH="/code"
-COPY custom_scripts/ /code/custom_scripts/
-COPY grinder/ /code/grinder/
-COPY map/ /code/map/
-COPY plugins/ /code/plugins/
-COPY queries/ /code/queries/
-COPY tests/ /code/tests/
-COPY grinder.py /code/grinder.py
-RUN mkdir /code/results/
-
-WORKDIR /code
-ENTRYPOINT ["python3", "/code/grinder.py"]
\ No newline at end of file
+# TLS Scanner build container --------------------------------------------------
+
+FROM maven:3-jdk-8 AS tls-scanner-build
+
+RUN git clone --depth=1 --branch '3.1' https://github.com/RUB-NDS/TLS-Attacker.git && \
+	git clone --depth=1 --recurse-submodules --branch '2.9' https://github.com/RUB-NDS/TLS-Scanner.git && \
+	(cd /TLS-Attacker/ && mvn clean install -DskipTests=true) && \
+	(cd /TLS-Scanner/ && mvn clean install -DskipTests=true)
+
+
+# Grinder ----------------------------------------------------------------------
+
+FROM python:3.7-alpine
+
+COPY . /app/
+COPY --from=tls-scanner-build /TLS-Scanner/apps /app/TLS-Scanner/apps
+
+RUN apk update && apk upgrade && \
+	apk add --no-cache nmap nmap-scripts && \
+	apk add --no-cache libpng freetype libstdc++ pkgconfig openjdk8 && \
+	apk add --no-cache --virtual .build-deps gcc build-base python-dev libpng-dev musl-dev freetype-dev && \
+	ln -s /usr/include/locale.h /usr/include/xlocale.h && \
+	pip install --no-cache-dir -r /app/requirements.txt && \
```

Comment:
Currently, we have "RuntimeError: Running cythonize failed!", we can fix it with the following line in Dockerfile:
`pip install --no-cache-dir cython`